nni/common/hpo_utils/validation.py

```diff
@@ -1,8 +1,10 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT license.
 
+from __future__ import annotations
+
 import logging
-from typing import Any, List, Optional
+from typing import Any
```

Comment:
we recommended using `list` not `List`, and `| None` not `Optional`?

Reply:
I personally prefer the new way. It think it looks much cleaner, and is closer to TypeScript style, which saves mental context switch overhead for me.

Reply:
If other modules choose to use 3.7 grammar, I think it's pretty okay and no need to change.
Notably `nni.experiment.config` will keep 3.7 grammar for years. Because it needs runtime checking and I'm afraid new grammar will not work.